phoenix-core/src/main/java/org/apache/phoenix/util/ScanUtil.java

```diff
@@ -1239,4 +1239,21 @@ public static boolean isDummy(List<Cell> result) {
         Cell cell = result.get(0);
         return CellUtil.matchingColumn(cell, EMPTY_BYTE_ARRAY, EMPTY_BYTE_ARRAY);
     }
+
+    public static void setWALAnnotationAttributes(PTable table, Scan scan) {
```

Comment:
@gjacoby126 Here is my understanding overall, please correct me if I'm wrong. There are 2 basic paths:

1. (Client-side delete) Client issues Puts/Deletes directly: This is handled in `MutationState. annotateMutationsWithMetadata()`
2. (Server-side delete and upsert-selects) Client issues scans, the server intercepts this and issues Put/Delete mutations based on the results read: This is what this method handles right? My understanding is we set the scan attributes in UpsertCompiler and DeleteCompiler and this is intercepted in `UngroupedAggregateRegionScanner`  where we read the scan attribute and set it on the mutations.

Reply:
Can we add comments based on this in the compiler and/or this method. Also a similar comment in MutationState would be helpful.

Reply:
Added comments to the Utility methods in ScanUtil and WALAnnotationUtil to avoid duplication